packages/hdwallet-keepkey/src/osmosis.ts

```diff
@@ -0,0 +1,293 @@
+          redelegate.setDelegatorAddress(m.value.delegator_address);
+          redelegate.setValidatorSrcAddress(m.value.validator_src_address);
+          redelegate.setValidatorDstAddress(m.value.validator_dst_address);
+          redelegate.setAmount(m.value.amount.amount);
+          redelegate.setDenom(m.value.amount.denom);
+
+          ack = new OsmosisMessages.OsmosisMsgAck();
+          ack.setRedelegate(redelegate);
+          break;
+        }
+        case "cosmos-sdk/MsgWithdrawDelegationReward": {
+          // Rewards
+          const rewards = new OsmosisMessages.OsmosisMsgRewards();
+          rewards.setDelegatorAddress(m.value.delegator_address);
+          rewards.setValidatorAddress(m.value.validator_address);
+
+          ack = new OsmosisMessages.OsmosisMsgAck();
+          ack.setRewards(rewards);
+          break;
+        }
+        case "osmosis/gamm/join-pool": {
+          // LP add
+          const lpAdd = new OsmosisMessages.OsmosisMsgLPAdd();
+          lpAdd.setSender(m.value.sender);
+          lpAdd.setPoolId(m.value.pool_id);
+          lpAdd.setShareOutAmount(m.value.share_out_amount);
+          lpAdd.setDenomInMaxA(m.value.token_in_maxs[0].denom);
+          lpAdd.setAmountInMaxA(m.value.token_in_maxs[0].amount);
+          lpAdd.setDenomInMaxB(m.value.token_in_maxs[1].denom);
+          lpAdd.setAmountInMaxB(m.value.token_in_maxs[1].amount);
+
+          ack = new OsmosisMessages.OsmosisMsgAck();
+          ack.setLpAdd(lpAdd);
+          break;
+        }
+        case "osmosis/gamm/exit-pool": {
+          // LP remove
+          const lpRemove = new OsmosisMessages.OsmosisMsgLPRemove();
+          lpRemove.setSender(m.value.sender);
+          lpRemove.setPoolId(m.value.pool_id);
+          lpRemove.setShareInAmount(m.value.share_in_amount);
+          lpRemove.setDenomOutMinA(m.value.token_out_mins[0].denom);
+          lpRemove.setAmountOutMinA(m.value.token_out_mins[0].amount);
+          lpRemove.setDenomOutMinB(m.value.token_out_mins[1].denom);
+          lpRemove.setAmountOutMinB(m.value.token_out_mins[1].amount);
+
+          ack = new OsmosisMessages.OsmosisMsgAck();
+          ack.setLpRemove(lpRemove);
+          break;
+        }
+        case "osmosis/lockup/lock-tokens": {
+          // LP stake
+          const lpStake = new OsmosisMessages.OsmosisMsgLPStake();
+          lpStake.setOwner(m.value.owner);
+          lpStake.setDuration(m.value.duration);
+          lpStake.setDenom(m.value.coins[0].denom);
+          lpStake.setAmount(m.value.coins[0].amount);
+
+          ack = new OsmosisMessages.OsmosisMsgAck();
+          ack.setLpStake(lpStake);
+          break;
+        }
+        case "osmosis/lockup/begin-unlock-period-lock": {
+          // LP unstake
+          const lpUnstake = new OsmosisMessages.OsmosisMsgLPUnstake();
+          lpUnstake.setOwner(m.value.owner);
+          lpUnstake.setId(m.value.id);
+
```

Comment:
also supports specific value if we wanted to add

Reply:
looks like both would actually require firmware update, so kind of irrelevant at this point

Reply:
LP token bonding support isn't going out with firmware release v7.7.0. We can remove the lpStake/unstake handlers from this PR or leave them as-is for now.

Reply:
If not supported by the firmware, I think it would be better to leave out imo.

Reply:
Done deal - removing now.